core/services/mongodb-remote/src/main/java/com/mongodb/stitch/core/services/mongodb/remote/sync/internal/DataSynchronizer.java

```diff
@@ -1109,16 +1153,6 @@ private void syncLocalToRemote() {
                   committedEvent.getUpdateDescription(),
                   false));
 
-          if (nextVersion == null) {
```

Comment:
@adamchel I couldn't tell if this was really necessary. I could add this back and check that OP is not DELETE. Thoughts?